core/src/main/java/org/fao/geonet/kernel/security/keycloak/KeycloakAuthenticationProcessingFilter.java

```diff
@@ -144,7 +146,18 @@ protected void successfulAuthentication(HttpServletRequest request, HttpServletR
                         } else {
                             Map<String, List<String>> qsMap = splitQueryString(request.getQueryString());
                             if (qsMap.containsKey("redirectUrl")) {
-                                response.sendRedirect(qsMap.get("redirectUrl").get(0));
+                                String redirectUrl = qsMap.get("redirectUrl").get(0);
+                                // redirectUrl should only be relative to the current server so remove the host from the url so that we don't allow redirects to other hosts.
+                                if (redirectUrl != null && !redirectUrl.startsWith("/")) {
```

Comment:
Could not be just checked if the url is absolute with `URL.isAbsolute()` and then just redirect back to the context home?

Reply:
@josegar74 - the returning url was being supplied as an absolute path.  But I guess there is no reason to supply an absolute path if we only want to use relative paths.

I have changed the code so that it supplied a relative path for the returning url and it also use URI.isAbsolute() for checking the redirectUrl parameter as you suggested.